src/docs/markdown/automatic-https.md

```diff
@@ -23,7 +23,7 @@ Here's a 28-second video showing how it works:
 - Caddy serves IP addresses and local/internal hostnames over HTTPS with locally-trusted certificates. Examples: `localhost`, `127.0.0.1`.
 - Caddy serves public DNS names over HTTPS with certificates from [Let's Encrypt](https://letsencrypt.org). Examples: `example.com`, `sub.example.com`, `*.example.com`.
 
-Caddy keeps all certificates renewed, and redirects HTTP (default port 80) to HTTPS (default port 443) automatically.
+Caddy keeps all certificates renewed, and redirects HTTP (default port 80) to HTTPS (default port 443) automatically, provided that [activation](/docs/automatic-https#activation) is successful.
```

Comment:
> This is basically the same thing as "Automatic HTTPS works as long as it does not fail," which seems obvious/redundant

Mostly the intent was to point the user to the Activation section for a list of criteria for when `auto_https` would work or fail. The original statement just implies it all works well with no clarification of why it might fail.

A site address of `localhost` by default implicitly connects over HTTPS with HTTP->HTTPS redirect, `off` connects to HTTP with no HTTPS or redirect, while `disable_redirects` results in HTTP not being reachable(not my original expectation), or `localhost:80` being reachable but no HTTPS anymore despite `auto_https disable_redirects` behaviour without the explicit port, and that turns out to be due to another non-obvious rule/condition at play.

So while it seems redundant, it's aimed at directing the user to the right docs when troubleshooting, they shouldn't need to go through the entire docs page looking for some hopeful explanation or worse trying to find such information elsewhere in the docs, especially if this feature spans multiple doc pages about it (which might not be known of at the time either).

---

Slight change of phrasing, as "Activation" personally didn't *click* as well when I first saw it. "rules" might also be appropriate alternative to "conditions":

```suggestion
When the [conditions for automatic HTTPS](/docs/automatic-https#activation) are valid, Caddy keeps all certificates renewed, and redirects HTTP (default port 80) to HTTPS (default port 443) automatically.
```

Reply:
"Conditions are valid" is weird/awkward. I still prefer my suggestion above, as "activation" is probably the best word for it.

Reply:
I feel the same way about "activation" personally :)

I'll go with your suggestion, I understand the doc better now that it doesn't bother me and isn't something worth debating further.